### ISSUE TITLE
Computing validation loss and validation examples

### ISSUE THREAD

**Issue by rom1504** [lucidrains/DALLE-pytorch#238]
Wouldn't it be good to compute a validation loss every N steps and an example image from a caption of a validation dataset ? (And report that to wandb)
To make sure no over fitting is happening and check the generalization capabilities
